partner_firstname/models/res_partner.py

```diff
@@ -236,3 +230,16 @@ def _install_partner_firstname(self):
         # Force calculations there
         records._inverse_name()
         _logger.info("%d partners updated installing module.", len(records))
+
+    @api.multi
+    def write(self, vals):
+        name = vals.get('name')
+        if name and all(name == partner.name for partner in self):
+            vals.pop('name', None)
+        # If vals is empty (only write name field and with the same value)
+        # Avoid access checking here
+        # https://github.com/odoo/odoo/blob/
+        #   8b83119fad7ccae9f091f12b6ac89c2c31e4bac3/openerp/addons/base/res/
+        #   res_partner.py#L569
+        this = self.sudo() if not vals else self
```

Comment:
@pedrobaeza I was with @antespi working on this and I know this is a workaround needed to avoid an access error when autosubscribing somebody. Odoo behaves weirdly here.

Anyway, I think we should write a test that tries to set the same name a partner has into itself, but written with a unprivileged user, and check no exceptions are raised.

This bug was very hard to locate, and I think this simple test is worth it to avoid future replications.

What do you think?